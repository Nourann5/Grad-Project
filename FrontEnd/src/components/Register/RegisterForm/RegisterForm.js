import React, { useEffect, useRef, useState } from 'react'
import {ReactComponent as StudentIcon} from 'assets/icons/studentIcon.svg'
import {ReactComponent as TeacherIcon} from 'assets/icons/teacherIcon.svg'
import {ReactComponent as SchoolIcon} from 'assets/icons/schoolIcon.svg'
import {ReactComponent as CenterIcon} from 'assets/icons/centerIcon.svg'
import {ReactComponent as CenterAcademy} from 'assets/icons/centerAcademy.svg'
import {ReactComponent as ColleageIcon} from 'assets/icons/colleageIcon.svg'
import {ReactComponent as UniversityIcon} from 'assets/icons/universityIcon.svg'

import styles from './RegisterForm.module.css'
import { NavLink } from 'react-router-dom'
import { Col, Row } from 'react-bootstrap'
import { useFormContext } from 'react-hook-form'
function RegisterForm({setCurrentActiveForm}) {
    const {register,getValues,watch} = useFormContext()
    useEffect(()=>{
        if(getValues('user_type')){
            setCurrentActiveForm(1)
        }
    },[watch('user_type')])
    return (
        <>
            <div className={styles["first-sign-up-form-content"]}>
                <div className={styles["first-sign-up__form-content-container"]}>
                    <p className={styles["first-sign-up-form-title"]}>Getting Started</p>
                    <p className={styles["first-sign-up-form-second-title"]}>Lorem ipsum dolor sit amet, consetetur sadipscin nsetetur sadipscing</p>
                    <Row className='justify-content-center'>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='student' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerStudentType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerStudentType' className={styles["first-sign-up__role"]}>
                                <StudentIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>Student</p>
                            </label>
                        </Col>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='teacher' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerTeacherType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerTeacherType' className={styles["first-sign-up__role"]}>
                                <TeacherIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>Teacher</p>
                            </label>
                        </Col>
                        {/* <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='school' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerSchoolType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerSchoolType' className={styles["first-sign-up__role"]}>
                                <SchoolIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>School</p>
                            </label>
                        </Col>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='center' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerCenterType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerCenterType' className={styles["first-sign-up__role"]}>
                                <CenterIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>Center</p>
                            </label>
                        </Col>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='academyCenter' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerCenterAcademyType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerCenterAcademyType' className={styles["first-sign-up__role"]}>
                                <CenterAcademy className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>Center Academy</p>
                            </label>
                        </Col>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='colleage' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerColleageType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerColleageType' className={styles["first-sign-up__role"]}>
                                <ColleageIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>Colleage</p>
                            </label>
                        </Col>
                        <Col lg='3' md='4' xs='6' className='mb-4 '>
                            <input
                                type='radio' 
                                value='university' 
                                className={styles['first-sign-up__role-input-radio']}
                                id='registerUniversityType'
                                {...register('user_type')} 
                            />
                            <label htmlFor='registerUniversityType' className={styles["first-sign-up__role"]}>
                                <UniversityIcon className={styles["first-sign-up__role-icon"]}/>
                                <p className={styles["first-sign-up__role-name"]}>University</p>
                            </label>
                        </Col> */}
                    </Row>
                </div>
            </div>
            <p className={styles["first-sign-up__alredy-have-accout"]}> Already Have An Account ? 
                <NavLink to='/login' className={styles["first-sign-up__move-to-login"]}> Log In</NavLink>
            </p>
        </>
    )
}

export default RegisterForm