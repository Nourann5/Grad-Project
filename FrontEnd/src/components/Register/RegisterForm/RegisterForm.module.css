.first-sign-up__form-head{
    display: flex;
    align-items: center;
    margin-bottom: 15px;
}
.first-sign-up__return-para{
    position: absolute;
    color:var(--text-color-dark);
    font-size:18px;
    font-family:'PoppinsMedium';
    transition:var(--transition);
}
.first-sign-up__return-para:hover{
    color:var(--main-color);
}
.first-sign-up__return-icon{
    margin-right: 5px;
    width: 20px;
    height: 20px;
}
.first-sign-up__logo{
    margin: auto;
    display:none
}
.first-sign-up__logo.active{
    display:initial;
}
.first-sign-up-form-content{
    width:60%;
    margin: auto;
    background-color:white;
    box-shadow: 10px 10px 35px #4807E61A;
    padding: 30px 0;
    margin-bottom: 30px;
    border-radius: 5px;
}
.first-sign-up__form-content-container{
    width:85%;
    margin: auto;
}
.first-sign-up-form-title{
    font-size: 26px;
    font-family: 'PoppinsSemiBold';
    color: var(--text-color-dark);
    margin-bottom: 20px;
    text-align: center;
}
.first-sign-up-form-second-title{
    font-size: 18px;
    font-family: 'PoppinsMedium';
    color: var(--text-color-dark);
    margin-bottom: 20px;
    text-align: center;
}
.first-sign-up__alredy-have-accout{
    text-align: center;
    color: #8B8B8B;
    font-size: 22px;
    font-family: 'poppinsMedium';
}
.first-sign-up__role{
    background-color: white;
    color: var(--text-color-dark);
    box-shadow: 6px 6px 45px #4807e612;
    border: 0;
    cursor: pointer;
    border-radius: 8px;
    width:100%;
    height:130px;
    transition:var(--transition) ;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}
.first-sign-up__role-name{
    font-size: 14px;
    font-family: 'PoppinsBold';
    transition:var(--transition);
    margin: 0;
}
.first-sign-up__role:hover .first-sign-up__role-name{
    color:white
}
.first-sign-up__role:hover{
    background-color:var(--main-color);
    color:var(--text-color-dark)
}
.first-sign-up__role-icon{
    border-radius: 50%;
    transition:var(--transition);
    background-color:#F4F7FA;
    margin-bottom: 10px;

}
.first-sign-up__role:hover .first-sign-up__role-icon{
    background-color:white
}
.first-sign-up__move-to-login{
    color: var(--main-color);
    font-size: 24px;
    font-family:'PoppinsSemiBold'
}
.first-sign-up__role-input-radio{
    display:none
}
@media(max-width:1024px){
    .first-sign-up-form-content{
        width:80%
    }
    .first-sign-up__form-content-container{
        width:80%
    }
}
@media(max-width:768px){
    .first-sign-up-form-content{
        width:90%
    }
    .first-sign-up__form-content-container{
        width:90%
    }
    #first-sign-up{
        padding: 30px 5px;
    }
}
@media(max-width:426px){
    .first-sign-up-form-content{
        padding: 20px 10px;
        margin-bottom: 5px;
    }
    #first-sign-up {
        padding: 10px 0px;
    }
    .first-sign-up-form-title {
        font-size: 22px;
    }
    .first-sign-up__return-span{
        display:none
    }
    .first-sign-up__return-icon {
        width: 40px;
        height: 40px;
        color: var(--main-color);
        margin-left: 10px;
    }
    .first-sign-up__move-to-login{
        font-size: 16px;
    }
    .first-sign-up__logo{
        width:70px;
        height:70px;
    }
}
