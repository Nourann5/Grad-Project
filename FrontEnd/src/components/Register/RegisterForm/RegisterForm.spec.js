import {render , renderHook, screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import { FormProvider, useForm } from 'react-hook-form'
import RegisterForm from './RegisterForm'

const methods = renderHook(()=> useForm())

describe('Test the Registeration Form ', () => { 
    it('Registeration Form',()=>{
        const TestInput = () => {
            const methods = useForm();
            return (
                <BrowserRouter>
                    <FormProvider {...methods}>
                        <RegisterForm />
                    </FormProvider>
                </BrowserRouter>
            );
        };
        const tree =  render(<TestInput/>)
        expect(tree).toMatchSnapshot()
    })
})