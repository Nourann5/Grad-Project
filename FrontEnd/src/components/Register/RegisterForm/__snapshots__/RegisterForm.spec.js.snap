// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Registeration Form  Registeration Form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="first-sign-up-form-content"
      >
        <div
          class="first-sign-up__form-content-container"
        >
          <p
            class="first-sign-up-form-title"
          >
            Getting Started
          </p>
          <p
            class="first-sign-up-form-second-title"
          >
            Lorem ipsum dolor sit amet, consetetur sadipscin nsetetur sadipscing
          </p>
          <div
            class="justify-content-center row"
          >
            <div
              class="mb-4  col-lg-3 col-md-4 col-6"
            >
              <input
                class="first-sign-up__role-input-radio"
                id="registerStudentType"
                name="user_type"
                type="radio"
                value="student"
              />
              <label
                class="first-sign-up__role"
                for="registerStudentType"
              >
                <svg
                  class="first-sign-up__role-icon"
                >
                  studentIcon.svg
                </svg>
                <p
                  class="first-sign-up__role-name"
                >
                  Student
                </p>
              </label>
            </div>
            <div
              class="mb-4  col-lg-3 col-md-4 col-6"
            >
              <input
                class="first-sign-up__role-input-radio"
                id="registerTeacherType"
                name="user_type"
                type="radio"
                value="teacher"
              />
              <label
                class="first-sign-up__role"
                for="registerTeacherType"
              >
                <svg
                  class="first-sign-up__role-icon"
                >
                  teacherIcon.svg
                </svg>
                <p
                  class="first-sign-up__role-name"
                >
                  Teacher
                </p>
              </label>
            </div>
          </div>
        </div>
      </div>
      <p
        class="first-sign-up__alredy-have-accout"
      >
         Already Have An Account ?
        <a
          class="first-sign-up__move-to-login"
          href="/login"
        >
           Log In
        </a>
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="first-sign-up-form-content"
    >
      <div
        class="first-sign-up__form-content-container"
      >
        <p
          class="first-sign-up-form-title"
        >
          Getting Started
        </p>
        <p
          class="first-sign-up-form-second-title"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscin nsetetur sadipscing
        </p>
        <div
          class="justify-content-center row"
        >
          <div
            class="mb-4  col-lg-3 col-md-4 col-6"
          >
            <input
              class="first-sign-up__role-input-radio"
              id="registerStudentType"
              name="user_type"
              type="radio"
              value="student"
            />
            <label
              class="first-sign-up__role"
              for="registerStudentType"
            >
              <svg
                class="first-sign-up__role-icon"
              >
                studentIcon.svg
              </svg>
              <p
                class="first-sign-up__role-name"
              >
                Student
              </p>
            </label>
          </div>
          <div
            class="mb-4  col-lg-3 col-md-4 col-6"
          >
            <input
              class="first-sign-up__role-input-radio"
              id="registerTeacherType"
              name="user_type"
              type="radio"
              value="teacher"
            />
            <label
              class="first-sign-up__role"
              for="registerTeacherType"
            >
              <svg
                class="first-sign-up__role-icon"
              >
                teacherIcon.svg
              </svg>
              <p
                class="first-sign-up__role-name"
              >
                Teacher
              </p>
            </label>
          </div>
        </div>
      </div>
    </div>
    <p
      class="first-sign-up__alredy-have-accout"
    >
       Already Have An Account ?
      <a
        class="first-sign-up__move-to-login"
        href="/login"
      >
         Log In
      </a>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
